default/AI/ResearchAI.py

```diff
@@ -443,6 +444,44 @@ def generate_research_orders():
                         idx = research_queue_list.index("SHP_WEAPON_4_2")
                         fo.issueEnqueueTechOrder("SHP_WEAPON_4_2", max(0, idx-18))
 
+    # TODO: Remove the following example code
```

Comment:
Seems to me this TODO should really be to refactor most  of this into one ore more functions to better support actual use cases.  It's fine/understandable that this initial example focused on MilShipDesigns, but of course we'll want to be more general.  Seems to me that we'd want the input to the main function to be a list of techs to be considered together (not sequentially as in this example), so that the AI could assess the use of a line of tech that still has multiple stages unresearched; this function would then return a dict of usefulness scores, one with respect to each of the main ship-related priority types (military, troops/invasions, colony, outpost, scouting).